### PR TITLE
Nomis/dsos 1659/cloudwatch db batch issue metric

### PR DESCRIPTION
**Pull request by robertiansweetman**
Watches the file output by oracle to report on batch failures or long running batches and supplies this via the collectd exec plugin to the local Cloudwatch agent as a metric. This can be used later to trigger alarms or other activities.

### CODE REVIEW

File: ansible/roles/amazon-cloudwatch-agent/tasks/redhat.yml [hunk 1]
```diff
@@ -60,7 +65,7 @@
       ansible.builtin.template:
         src: oracle_health.sh.j2
         dest: /opt/collectd/oracle_health.sh
-        mode: 0500
+        mode: 0755
         owner: oracle
 
     - name: Make Oracle Health Check script executable
```
Comment (drobinson-moj) [inline ansible/roles/amazon-cloudwatch-agent/tasks/redhat.yml:71]: I guess this step is redundant now as the above step copies with 0755 permissions

File: ansible/roles/amazon-cloudwatch-agent/handlers/main.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+- name: restart collectd
+  ansible.builtin.service:
+    name: collectd
+    state: restarted
+    enabled: yes
```
Comment (drobinson-moj) [inline ansible/roles/amazon-cloudwatch-agent/handlers/main.yml:6]: If the purpose of the handler is to restart collectd, I would just do that and not enable it as well.  (Although to be fair, other roles do the same)

File: ansible/roles/amazon-cloudwatch-agent/tasks/redhat.yml [hunk 3]
```diff
@@ -72,6 +77,7 @@
         name: collectd
         state: started
         enabled: yes
+      notify: restart collectd
```
Comment (drobinson-moj) [inline ansible/roles/amazon-cloudwatch-agent/tasks/redhat.yml:80]: I'd normally expect to see this notify under steps which are changing the collectd configuration files and require a restart.  I think this step can just start and enable the service

**Review by drobinson-moj — APPROVED** [review]
Approved - but some comments above

File: ansible/roles/amazon-cloudwatch-agent/tasks/redhat.yml (hunk 3 above)
Comment (robertiansweetman) [reply ansible/roles/amazon-cloudwatch-agent/tasks/redhat.yml:80]: I was thinking about if the role is re-run on the basis that the collectd config 'probably' changed then this needs restarting to pick up the new config. I've not found an auto-load option in collectd yet

File: ansible/roles/amazon-cloudwatch-agent/tasks/redhat.yml (hunk 1 above)
Comment (robertiansweetman) [reply ansible/roles/amazon-cloudwatch-agent/tasks/redhat.yml:71]: Yep, that'll be me not having comprehended permissions quite yet +1

File: ansible/roles/amazon-cloudwatch-agent/tasks/redhat.yml (hunk 3 above)
Comment (drobinson-moj) [reply ansible/roles/amazon-cloudwatch-agent/tasks/redhat.yml:80]: If you re-run, it won't trigger the notify as this step will be green/unchanged, i.e. the service is already started and enabled.  So you need to put the notify on steps which actually change the collectd config

Comment (robertiansweetman) [reply ansible/roles/amazon-cloudwatch-agent/tasks/redhat.yml:80]: Ah, I was wondering about that - will change it elsewhere when I start getting the windows config agent working :+1

**Pull request merged**